etc/bin/update-deps.sh

```diff
@@ -0,0 +1,175 @@
+# unset, we set the corresponding environment variables. We use information
+# from BuildKite when available and fall back to GitHub user information if
+# it's unavailable.
+if ! git config user.name >/dev/nul; then
+  export GIT_AUTHOR_NAME="${BUILDKITE_BUILD_CREATOR:-$GITHUB_USER}"
+  export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+fi
+
+if ! git config user.email >/dev/null; then
+  FALLBACK_EMAIL="$GITHUB_USER@users.noreply.github.com"
+  export GIT_AUTHOR_EMAIL="${BUILDKITE_BUILD_CREATOR_EMAIL:-$FALLBACK_EMAIL}"
+  export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+fi
+
+# When pushing over ssh, automatically add the host to known_hosts instead of
+# prompting with,
+#
+#   The authenticity of host '...' can't be established.
+#   RSA key fingerprint is ...
+#   Are you sure you want to continue connecting (yes/no)?
+export GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no"
+
+# Returns the current date in ISO8601 format.
+now() {
+  date +%Y-%m-%dT%H:%M:%S
+}
+
+# Use this instead of calling `git status` directly.
+git_status()
+{
+  # BuildKite's docker-compose plugin generates a fake docker-compose so we
+  # need to ignore it anytime we do git status.
+  git status "$@" | grep -v '?? docker-compose.buildkite'
+}
+
+if [ -n "$(git_status --porcelain)" ]; then
+  echo "Working tree is dirty."
+  echo "Please verify that you don't have any uncommitted changes."
+  git status
+  exit 1
+fi
+
+echo "--- Updating dependencies"
+make glide-up
+
+case "$(git_status --porcelain)" in
+  "")
+    echo "Nothing changed. Exiting."
+    exit 0
+    ;;
+  " M glide.lock")
+    echo "--- glide.lock changed"
+    # Keep going
+    ;;
+  *)
+    echo "Unexpected changes after a glide up:"
+    git_status
+    exit 1
+esac
+
+git add glide.lock
+git commit -m "Update dependencies at $(now)"
+
+echo "--- Updating generated code"
+make generate
+
+# We want to push directly to the remote only if the generated code did not
+# change and all tests pass.
+if [ -z "$(git_status --porcelain)" ]; then
+  if make lint test examples; then
+    echo "--- Generated code did not change and the tests passed."
+    echo "--- Pushing changes and exiting."
+    git push "$REMOTE" HEAD:"$BRANCH"
+    exit 0
+  fi
+else
+  # Check in the generated code ignoring the BuildKite docker-compose.
+  git add -A
+  git rm --cached docker-compose.buildkite*
+  git commit -m "Update generated code at $(now)"
+fi
+
+PR_BRANCH=""
+if [ -z "${BUILDKITE_BUILD_NUMBER:-}" ]; then
+  # Use a different branch namespace if we're not running in BuildKite.
+  PR_BRANCH="update-deps/$(now)"
+else
+  PR_BRANCH="update-deps/buildkite/$BUILDKITE_BUILD_NUMBER"
+fi
+
+echo "--- Creating a pull request using branch $PR_BRANCH"
+git push "$REMOTE" HEAD:"refs/heads/$PR_BRANCH"
+
+MESSAGE=$(echo 'I tried to update the dependencies but either the generated
+code changed or some tests failed, so I need someone to validate or fix this
+change.
+
+Thanks!' | perl -p -e 's/\n/\\n/g')
```

Comment:
could use `tr` for this

Reply:
That was my first choice. `tr` does a 1-1 mapping between characters. You can't make it write two characters for one input character (to my knowledge).